### PR TITLE
#57 target platforms support

### PR DESCRIPTION
**Pull request by Olyve**
Still a work in progress, but looking for feedback

### CODE REVIEW

File: injected/src/services/plugins.ts [hunk 1]
```diff
@@ -7,6 +7,16 @@ interface Entrypoint {
   menu: boolean;
 }
 
+export interface StorePlatforms {
+  linux: Platform;
+  windows: Platform;
+  mac: Platform;
```
Comment (Olyve) [inline injected/src/services/plugins.ts:13]: mac is returned from the function as `darwin`. Probably should make a translation method to translate the platform into something we want, or encode the translation as an enum

**Comment by Olyve** [comment]
Need to rethink the typing here. Might be able to use an enum to store Platform info which would solve the translation between what we refer to platform os' as and how node refers to them.

**Comment by coolavery** [comment]
You could expose the platform in a global variable (like `window.csSteamDir`), it'll probably be useful once there are plugins for different platforms. It could also be a good place to centralize the platform parsing you mentioned for Mac and Windows.

For reference, the script template that defines those globals is [here](https://github.com/CrankshaftForSteam/crankshaft/blob/main/build/eval.template.js), and it gets filled in from [here](https://github.com/CrankshaftForSteam/crankshaft/blob/89a31e8dc6c15677efec566dc424342cd90774a1/build/build.go#L123-L139).

That template goes through Esbuild so I think you can still use `os.type`, or you can get it from go with `runtime.GOOS` which might simplify parsing (it should return `windows`, `darwin`, or `linux`).

**Comment by Olyve** [comment]
There is a lot of repeated code in `inject.go` and `app-properties` where we call the build script eval stuff. I'd like to figure out a way to simplify some of that at some point.

File: build/build.go [hunk 2]
```diff
@@ -105,38 +116,35 @@ func BundleSharedScripts() (string, error) {
...
+	globals := csGlobals{
 		Version:        VERSION,
 		InjectedScript: script,
 		ServerPort:     serverPort,
 		UIMode:         uiMode,
 		SteamDir:       steamPath,
 		AuthToken:      authToken,
 		PluginsDir:     pluginsDir,
-	}); err != nil {
+		Platform:       platform,
```
Comment (coolavery) [inline build/build.go:144]: I think we can directly set `Platform` here so we don't have to pass it in from every build call

```suggestion
		Platform:       runtime.GOOS,
```

**Review by coolavery — APPROVED** [review]

**Comment by coolavery** [comment]
Thanks! Merged manually:

> To git.sr.ht:~avery/crankshaft
>    720c18c..e2d3a60  main -> main

**Pull request closed without merging**